### PR TITLE
Add claim-neurons command, for claiming GTC seed neurons

### LINKED ISSUE #84: Add ability to send a claim message to the GTC canister
This involves performing the following call in the `gtc.did`:
```
service : {
  claim_neurons : (text) -> (Result);
}
```

The argument here is a hex string encoded public key. The same public key that the whole request is signed for.

An example of calling it from `dfx` is here:

https://sdk.dfinity.org/docs/token-holders/seed-donations.html

### PR DESCRIPTION
**Pull request by jwiegley**
Fixes #84

### CODE REVIEW

**Comment by jwiegley** [comment]
Fixes #84

**Comment by chmllr** [comment]
FYI I've merged master into this branch

**Review by chmllr — DISMISSED** [review]

File: candid/gtc.did [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+type Result_1 = variant { Ok; Err : text };
+type Result_2 = variant { Ok : AccountState; Err : text };
+type TransferredNeuron = record {
+  error : opt text;
+  timestamp_seconds : nat64;
+  neuron_id : opt NeuronId;
+};
+service : {
+  balance : (text) -> (nat32) query;
+  claim_neurons : (text) -> (Result);
```
Comment (timohanke) [inline candid/gtc.did:23]: We only need this line and the types it uses (Result, NeuronId). Can we remove all the rest from this file?

File: src/commands/claim_neurons.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    let private_key = EcKey::private_key_from_pem(pem.as_bytes())?;
+    let group = private_key.group();
+    let public_key = EcKey::from_public_key(group, private_key.public_key())?;
+    let mut context = BigNumContext::new()?;
+    let bytes = public_key.public_key().to_bytes(
+        public_key.group(),
+        PointConversionForm::UNCOMPRESSED,
+        &mut context,
+    )?;
+    let sig = Encode!(&hex::encode(&bytes))?;
```
Comment (timohanke) [inline src/commands/claim_neurons.rs:20]: Why is this called `sig` if it is actually the argument?

**Review by timohanke — CHANGES_REQUESTED** [review]
Can we strip down the gtc.did to only the necessary content?

**Comment by chmllr** [comment]
@timohanke I think it should work. It just would make it a bit harder to update the file in the future. But I doubt we plan to update the GTC cnaister. WDYT?

**Comment by timohanke** [comment]
> @timohanke I think it should work. It just would make it a bit harder to update the file in the future. But I doubt we plan to update the GTC cnaister. WDYT?

I think the point of stripping down the file is exactly that we do _not_ have to change the file if the GTC canister upgrades. Canister upgrades should be backwards compatible. If methods are added this doesn't affect quill which is only calling `claim_neurons`.

If we leave the full file here and the GTC upgrades then the files will differ. Even though the mismatch doesn't break anything it is at least weird.

**Comment by chmllr** [comment]
> the point of stripping down the file is exactly that we do not have to change the file if the GTC canister upgrades. 

If it upgrades in the backwards compatible way we do not need to change anything one way or another right? Only if the method we keep will upgrade in a non-backwards compatible way, we'd have a small hassle of upgrading the did file. But that's what I assumed would never happen: `But I doubt we plan to update the GTC cnaister`. 

It sounds like we're talking about the same, so there is no harm in stripping the file down 👍

**Review by timohanke — DISMISSED** [review]

**Review by chmllr — APPROVED** [review]

**Pull request merged**